### PR TITLE
[CWS] Add bind to activity dumps

### PR DESCRIPTION
**Pull request by spikat**
### What does this PR do?

This PR add bind events to activity dumps for all the output formats (json, msgp, graph and profile), at runtime as well as snapshots.

The result is a new section on activity dump, which looks like:
```
"sockets": [
  {
    "family": "AF_INET",
    "bind": {
      "port": 4242,
      "ip": "0.0.0.0"
    }
  }
]
```

### Motivation

Adding bound sockets to activity dumps will allow to enforce which binds a process could do with future security profiles.

### Additional Notes

It will be interesting to add connect events in the same way.

### Describe how to test/QA your changes

To QA:
- bind functional tests have been updated, you could run them
- You'll have to generate dumps and validates its content for every 4 formats.
- And .. for every usecases:
- RUNTIME: starting an activity dump for a specific comm (like nc), then bind a socket (like with)
- SPANSHOT: bind the socket before starting the AD
- CGROUP: start a docker and bind a socket

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by djmitche — APPROVED** [review]
pkg/config/config.go 👍🏽

File: docs/cloud-workload-security/secl.json [hunk 1]
```diff
@@ -644,7 +644,7 @@
           "name": "bind.addr.family",
           "type": "int",
           "definition": "Address family",
-          "constants": "Network Address Family constants"
```
Comment (paulcacheux) [inline docs/cloud-workload-security/secl.json:647]: Why was this removed ?

File: pkg/security/secl/model/model.go [hunk 2]
```diff
@@ -826,8 +826,9 @@ type DNSEvent struct {
 type BindEvent struct {
 	SyscallEvent
 
-	AddrFamily uint16        `field:"addr.family" constants:"Network Address Family constants"` // Address family
-	Addr       IPPortContext `field:"addr"`                                                     // Bound address
+	Addr       IPPortContext `field:"addr"`          // Bound address
+	AddrFamily uint16        `field:"addr.family"`   // Address family
+	Protocol   uint8         `field:"addr.protocol"` // Socket protocol
```
Comment (paulcacheux) [inline pkg/security/secl/model/model.go:831]: Let's start with not exporting this field

File: pkg/security/probe/probe.go [hunk 3]
```diff
@@ -1458,6 +1458,8 @@ func AppendProbeRequestsToFetcher(constantFetcher constantfetch.ConstantFetcher,
 	// network related constants
 	constantFetcher.AppendOffsetofRequest("net_device_ifindex_offset", "struct net_device", "ifindex", "linux/netdevice.h")
 	constantFetcher.AppendOffsetofRequest("sock_common_skc_net_offset", "struct sock_common", "skc_net", "net/sock.h")
+	constantFetcher.AppendOffsetofRequest("proto_name_offset", "struct proto", "name", "net/sock.h")
```
[the remark this answers is not in the document]
Comment (spikat) [reply pkg/security/probe/probe.go:1461]: done, thanks

**Review by buraizu — APPROVED** [review]
Looks fine from a docs perspective

File: pkg/security/secl/model/model.go (hunk 2 above)
Comment (spikat) [reply pkg/security/secl/model/model.go]: I've split the PR in two, the addition of the protocol to bind is now separated.
Thanks

**Review by lebauce — APPROVED** [review]

**Review by Gui774ume — APPROVED** [review]
